package-res/resources/web/pentaho/type/_type.js

```diff
@@ -0,0 +1,1362 @@
+          delete this._isBrowsable;
+        }
+      },
+      //endregion
+
+      //region isAdvanced property
+      // @type boolean
+      // -> boolean, Optional(false), Inherited, Configurable
+      // null || undefined -> reset
+      _isAdvanced: false,
+
+      /**
+       * Gets or sets the `isAdvanced` attribute of this type.
+       *
+       * Types with `isAdvanced` attributes set to `false` are typically immediately accessible to the user.
+       * An advanced type typically escapes the expected flow of utilization, yet it is
+       * sufficiently relevant to be shown in a user interface.
+       *
+       * Setting to a {@link Nully} value causes this attribute to use the inherited value,
+       * except for the root type, `Instance.type` (which has no ancestor), where the attribute is `false`.
+       *
+       * @type {boolean}
+       * @see pentaho.type.Type#isBrowsable
+       */
+      get isAdvanced() {
+        return this._isAdvanced;
+      },
+
+      set isAdvanced(value) {
+        if(value == null) {
+          this._resetIsAdvanced();
+        } else {
+          this._isAdvanced = !!value;
+        }
+      },
+
+      _resetIsAdvanced: function() {
+        if(this !== _type) {
+          delete this._isAdvanced;
+        }
+      },
+      //endregion
+
+      //region styleClass property
+      // @type nonEmptyString
+      // -> nonEmptyString, Optional(null), Configurable, Localized
+      // "" or undefined -> null conversion
+
+      _styleClass: null,
+
+      /**
+       * Gets or sets the CSS class associated with this type.
+       *
+       * This attribute is typically used to associate an icon with a type.
+       *
+       * Attempting to set to a non-string value type implicitly
+       * converts the value to a string before assignment.
+       *
+       * An empty string or `undefined` value is interpreted as `null`.
+       *
+       * @type {?nonEmptyString}
+       */
+      get styleClass() {
+        return this._styleClass;
+      },
+
+      set styleClass(value) {
+        // undefined or "" -> null conversion
+        this._styleClass = nonEmptyString(value);
+      },
+
+      // TODO: implement inheritedStyleClasses
+      /**
+       * Gets the style classes of this and any base types.
+       *
+       * @type string[]
+       * @readonly
+       */
+      get inheritedStyleClasses() {
+        /* istanbul ignore next: not implemented yet */
+        throw error.notImplemented("Implement me!");
+      },
+      //endregion
+
+      //region ordinal property
+      // @type integer
+      // -> Optional(0), Inherited, Configurable
+      _ordinal: 0,
+
+      /**
+       * Gets or sets the ordinal associated with this type.
+       *
+       * The ordinal is used to disambiguate the order with which a type (or an instance of it)
+       * is shown in a user interface.
+       *
+       * Setting to a {@link Nully} value causes this attribute to use the inherited value,
```

Comment:
Suggest:

Setting this to a {@link Nully} value causes this attribute to use the inherited value,